File: backend/game/RhelGamePython/chapter1_local_storage/challenge5.py
import fdiskGuide
import ChallengeGenerator


challenge_5 = ChallengeGenerator.ChallengeGenerator(
    quest_number=5,
    description="Master Partition Creation with fdisk",
    story="""\nAs you further explore storage management, you realize the need to partition available storage space to organize data effectively.
Partitioning allows for better utilization of storage resources and facilitates the management of data.
To accomplish this task, you embark on a quest to create partitions using the fdisk utility.\n
With determination, you assess the storage devices available for partitioning and the desired partition layout.
You consider factors such as partition sizes, types, and mount points, ensuring a well-organized storage structure.\n
Issuing commands to launch fdisk and create partitions, you proceed methodically, specifying the parameters for each partition.
As the commands execute, you carefully define the partition layout, adhering to best practices for storage management.\n""",
    options=[
        {
            'name': 'View fdisk Guide',
            'action': '',
            'function': fdiskGuide.fdisk_guide,
            'success_message': '',
        },
        {
            'name': 'Create DOS Partition Guide',
            'action': '',
            'function': fdiskGuide.create_dos_partition_guide,
            'success_message': '',
        },
        {
            'name': 'Create LVM Partition Guide',
            'action': '',
            'function': fdiskGuide.create_lvm_partition_guide,
            'success_message': '',
        },
        {
            'name': 'Create Swap Partition Guide',
            'action': '',
            'function': fdiskGuide.create_swap_partition_guide,
            'success_message': '',
        },
        {
            'name': 'Delete Partition Guide',
            'action': '',
            'function': fdiskGuide.delete_partition_guide,
            'success_message': '',
        },
    ]
)

File: backend/data/chapters/chapter1/challenges/challenge_5.js
import fdisk_guides from './missions/fdisk_guides.js';

const challenge_5 = {
  id: "challenge_5", 
  title: "Master Partition Creation with fdisk",
  story: `As you further explore storage management, you realize the need to partition available storage space to organize data effectively.
Partitioning allows for better utilization of storage resources and facilitates the management of data.
To accomplish this task, you embark on a quest to master the fdisk utility.

With determination, you assess the storage devices available for partitioning and the desired partition layout.
You consider factors such as partition sizes, types, and mount points, ensuring a well-organized storage structure.

The fdisk utility represents one of the most fundamental tools in Linux system administration. Through its command-line interface, you can create, modify, and delete partitions with precision. Understanding fdisk is essential for any administrator who works with storage systems.

This challenge will guide you through the essential partition management techniques, from basic operations to advanced partition type configurations.`,
  briefing: `Master the fdisk utility through comprehensive guides and interactive examples.`,
  prompt: `Explore the fdisk guides to understand partition management:`,
  options: fdisk_guides,
}

export default challenge_5;